CMSgov/ato-blueprint
# Feature/new project model and create project form

Branched off of `feature/project-model` associated with this PR: https://github.com/CMSgov/ato-blueprint/pull/61

Stories:
- https://jiraent.cms.gov/browse/ISPGBSS-755 (new project model)
- https://jiraent.cms.gov/browse/ISPGBSS-729 (new create project form)

### Changes
Pertaining to ISPGBSS-755 (new project model)
- Added a project directory to house a model, view, urls file for project
- Created a new project model which is temporarily called `package` to limit interference with the existing project model. _Note: the rename should happen when the former project model is removed._
    - This created two tables in the database: `project.package` and `project_package_components`. The latter was automatically created to handle the "many to many relationship" between projects and components
- Added urls for project pages as placeholders (per [this spreadsheet](https://docs.google.com/spreadsheets/d/18zvZJb3mQf3f1OHGLVbR6JrRV62rdEL_BporHc_QXwc/edit?pli=1#gid=0))
- Added a view file to reference the existing project.html
- Created tests in `project/tests/test_models.py`
- added `/project` directory to precommit 

Pertaining to ISPGBSS-729 (create new project form)
- minor project model updates
- url updates to handle new templates
- added `project-create` template for new project creation form
- added `project-create-success` template as a placeholder until we decide what should be on the landing page _(currently set to redirect to the "/projects" page upon form completion)_
- added `project-home` template
- added `package_navigation` function in utils 
- style addiitons
- Created tests in `project/tests/test_views.py` for the project homepage and the create project page

### Testing
Manual testing
1. pull the branch locally and load the homepage
2. navigate to the new project form via the url "/project/create"
3. verify the following:
    - back button redirects back to homepage
    - next button submits the form and redirects to /projects
    - filling in the form with a name, acronym, location, and impact level allows form submission
    - leaving any of the fields blank prevents form submission
    - check your local database's `project-package` table after form submission to verify the entry in the database matches your inputs
    - you can visit the project homepage for the project you just submitted using the id found in the database _(Example: http://localhost:8000/project/{id}/)_
    - form appearance matches the [Figma](https://www.figma.com/file/D6GFeen0aPc5QMFLG75t5T/Blueprint---UI-Screens?node-id=610%3A599)

Automated testing
1. Bash into the container `docker exec -it govready-q-dev bash`
    - Run new tests `python3 manage.py test project.tests`
    - Run all tests `python3 manage.py test`

### Developer Notes
Future work will be needed for:
- Confirmation Landing page
- Projects page
- Individual project page
- Add component to a project flow
- Create new a new model for components (related work, but not necessarily dependent)
- Cleaning up: removing the old project model and related code, renaming the new project model from "Package" to "Project"
- Need to investigate a good pattern to display extra text with choices coming from a django model (pertaining to impact level details showing up on the form- since this isn't a straightforward thing, it'll be done in another story)
- Need to add feature flagging at some point to handle the switchover between the old project model and new project model
